### PR TITLE
Select dark and light mode

### PR DESCRIPTION
**Pull request by manuelserafin**

### CODE REVIEW

File: src/routes/routes.tsx
Comment (franpossetto) [inline src/routes/routes.tsx:1]: Why did indentation change?

File: src/pages/map/map/Map.tsx [hunk 1]
```diff
@@ -27,9 +27,19 @@ export default function Map() {
     libraries: ["places"],
   });
 
+  const { style } = useMapStyle();
+
   useEffect(() => {
-    loader.load().then((google) => {
+    const loadMap = async () => { // Define una función asincrónica dentro de useEffect
```
Comment (franpossetto) [inline src/pages/map/map/Map.tsx]: Delete comments.

File: src/pages/map/map/Map.tsx [hunk 2]
```diff
@@ -27,9 +27,19 @@ export default function Map() {
...
+      const google = await loader.load(); // Espera a que el Loader cargue la API de Google Maps
+  
       if (mapRef.current) {
+        // Utiliza import() para cargar dinámicamente el archivo de estilos
+        const mapStyles = style === 'dark'
```
Comment (franpossetto) [inline src/pages/map/map/Map.tsx:35]: can you move this out of the useEffect?

File: src/pages/map/map/Map.tsx [hunk 3]
```diff
@@ -40,15 +50,18 @@ export default function Map() {
           mapTypeControl: false,
           zoomControl: false,
           fullscreenControl: false,
-          styles: darkMapStyles,
+          styles: mapStyles, // Aplica los estilos cargados
```
Comment (franpossetto) [inline src/pages/map/map/Map.tsx]: Delete comments

File: src/routes/routes.tsx [hunk 4]
```diff
@@ -8,74 +8,75 @@ import { Profile } from "../pages/profile/Profile";
 import { MapView } from "../pages/map/MapView";
 import { PlaceController } from "../contexts/PlaceContext";
 import Layout from "../components/layout/Layout";
-import PlaceTable from "../components/places/placeTable/PlaceTable";
+// import PlaceTable from "../components/places/placeTable/PlaceTable";
```
Comment (franpossetto) [inline src/routes/routes.tsx:11]: Why does placeTable is commented?

File: tailwind.config.js [hunk 5]
```diff
@@ -1,6 +1,7 @@
 /** @type {import('tailwindcss').Config} */
 export default {
   content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
+  darkMode: "class",
```
Comment (franpossetto) [inline tailwind.config.js:4]: What does this line mean?

File: src/contexts/MapStyleContext.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React, { createContext, useContext, useState } from 'react';
```
Comment (franpossetto) [inline src/contexts/MapStyleContext.tsx]: compare this context with the two contexts we already have. Use the same structure and naming convention

**Review by franpossetto — CHANGES_REQUESTED** [review]

File: src/pages/map/map/Map.tsx (hunk 2 above)
Comment (manuelserafin) [reply src/pages/map/map/Map.tsx:35]: The style variable comes from the context, which can change during the life cycle of the component. By placing mapStyles inside the useEffect, we ensure that it is recalculated each time the style changes.

File: src/pages/map/map/Map.tsx (hunk 1 above)
Comment (manuelserafin) [reply src/pages/map/map/Map.tsx]: Done

File: src/pages/map/map/Map.tsx (hunk 3 above)
Comment (manuelserafin) [reply src/pages/map/map/Map.tsx]: done

File: src/routes/routes.tsx (hunk 4 above)
Comment (manuelserafin) [reply src/routes/routes.tsx:11]: 'PlaceTable' is declared but its value is never read. Do i delete de line?

File: tailwind.config.js (hunk 5 above)
Comment (manuelserafin) [reply tailwind.config.js:4]: That line is what makes the magic! That line is necessary for Tailwin to understand the change to dark mode.

File: src/routes/routes.tsx (hunk 4 above)
Comment (franpossetto) [reply src/routes/routes.tsx:11]: yes please

Comment (manuelserafin) [reply src/routes/routes.tsx:11]: done

File: src/contexts/MapStyleContext.tsx (hunk 6 above)
Comment (manuelserafin) [reply src/contexts/MapStyleContext.tsx]: done

**Pull request merged**